src/persistentSubscription/utils/persistentSubscriptionSettings.ts

```diff
@@ -84,21 +77,51 @@ export interface PersistentSubscriptionSettings {
   strategy: ConsumerStrategy;
 }
 
-export const persistentSubscriptionSettingsFromDefaults = (
-  changes: Partial<PersistentSubscriptionSettings> = {}
-): PersistentSubscriptionSettings => ({
+export interface PersistentSubscriptionSettings
+  extends PersistentSubscriptionSettingsGeneric {
+  /**
+   * The exclusive position in the stream or transaction file the subscription should start from.
+   * @default END
+   */
+  startFrom: typeof START | typeof END | bigint;
+}
+
+export interface PersistentSubscriptionToAllSettings
+  extends PersistentSubscriptionSettingsGeneric {
+  /**
+   * The exclusive position in the stream or transaction file the subscription should start from.
+   * @default END
+   */
+  startFrom: typeof START | typeof END | Position;
+}
+
+const defaults: PersistentSubscriptionSettingsGeneric = {
   resolveLinkTos: false,
-  extraStats: false,
-  fromRevision: START,
+  extraStatistics: false,
   messageTimeout: 30_000,
   maxRetryCount: 10,
   checkpointAfter: 2_000,
-  minCheckpointCount: 10,
-  maxCheckpointCount: 1_000,
-  maxSubscriberCount: UNLIMITED,
+  checkPointLowerBound: 10,
+  checkPointUpperBound: 1_000,
+  maxSubscriberCount: UNBOUNDED,
   liveBufferSize: 500,
   readBatchSize: 20,
   historyBufferSize: 500,
   strategy: ROUND_ROBIN,
```

Comment:
**SUGGESTION:** As we're aligning terminology, then we could also change it to `consumerStrategyName`.
```suggestion
  consumerStrategyName: ROUND_ROBIN,
```